### PR TITLE
Room list

### PR DESCRIPTION
**Pull request by yetieaterxb1**
# Submit a pull request

## CLA

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [ ] The code changes follow best practices
- [ ] Code changes are tested (add some information if not applicable)

### CODE REVIEW

File: examples/website-demos/virtual-event/src/components/CentralPanel/EventCard.tsx [hunk 1]
```diff
@@ -1,15 +1,16 @@
 import './EventCard.scss';
 import { EventCardIcon, ParticipantsIcon, VideoViewersIcon } from '../../assets';
 import { useEventContext } from '../../contexts/EventContext';
+import React from 'react';
```
Comment (DanC5) [inline examples/website-demos/virtual-event/src/components/CentralPanel/EventCard.tsx:4]: react import should always be first

File: examples/website-demos/virtual-event/src/components/CentralPanel/EventCard.tsx [hunk 2]
```diff
@@ -27,9 +28,7 @@ export const EventCard: React.FC<Props> = (props) => {
 
   return (
     <div className='event-card-container' onClick={handleClick}>
-      <div className='event-card-image'>
-        <Image />
-      </div>
+      <div className='event-card-image'>{Image && <Image />}</div>
```
Comment (DanC5) [inline examples/website-demos/virtual-event/src/components/CentralPanel/EventCard.tsx:31]: evaluate the Image conditional on the left side of the div, otherwise you're passing `false` to the div if it doesnt exist

File: examples/website-demos/virtual-event/src/components/CentralPanel/rooms.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    content:
+      'How to set a business plans to use information to a competitive advantage and support enterprise goals.',
+    eventName: 'qa',
+    Image: Card5,
+    label: 'Moderated',
+    presenters: 2,
+    title: 'Q&A session: Data strategy and executive communication',
+    viewers: 86,
+  },
+
```
Comment (DanC5) [inline examples/website-demos/virtual-event/src/components/CentralPanel/rooms.ts:51]: event names and content are identical here

File: examples/website-demos/virtual-event/src/components/Chat/DMChannel.scss [hunk 4]
```diff
@@ -17,6 +17,7 @@
       transition: background 1s;
       align-items: center;
       justify-content: space-between;
+      width: 100%;
```
Comment (DanC5) [inline examples/website-demos/virtual-event/src/components/Chat/DMChannel.scss:20]: why was this addition needed here?

File: examples/website-demos/virtual-event/src/components/CentralPanel/rooms.ts (hunk 3 above)
Comment (yetieaterxb1) [reply examples/website-demos/virtual-event/src/components/CentralPanel/rooms.ts:51]: Yeah the cards are the exact same

File: examples/website-demos/virtual-event/src/components/CentralPanel/RoomsList.tsx [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+                  </div>
+                );
+              })}
+            </div>
+          )}
+        </div>
+        <div className='rooms-list-container-cards'>
+          {rooms.map((room) => {
+            return (
+              <div className='rooms-list-card'>
```
Comment (DanC5) [inline examples/website-demos/virtual-event/src/components/CentralPanel/RoomsList.tsx:57]: put this div inside the EventCard component and implicit return the items

File: examples/website-demos/virtual-event/src/components/CentralPanel/RoomsList.tsx [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+  return (
+    <>
+      <div className={`rooms-list-video${chatType === 'room' ? '' : '-hidden'}`}>
+        <h1>ROOM VIDEO</h1>
+      </div>
+      <div className={`rooms-list-container${chatType === 'room' ? '-hidden' : ''}`}>
+        <div className='rooms-list-header'>
+          <div className='rooms-list-header-title'>
+            <div className='rooms-list-header-title-main'>World Hacker Summit 2021</div>
+            <div className='rooms-list-header-title-sub'>Stream.IO</div>
```
Comment (DanC5) [inline examples/website-demos/virtual-event/src/components/CentralPanel/RoomsList.tsx:29]: i know this in the designs but we're not stream.io, maybe chance to presented by Stream

File: examples/website-demos/virtual-event/src/components/CentralPanel/RoomsList.tsx [hunk 7]
```diff
@@ -0,0 +1,75 @@
+import { useState } from 'react';
+
+import { EventCard } from './EventCard';
+import { rooms } from './rooms';
+
+import { CalendarButton } from '../../assets/CalendarButton';
```
Comment (DanC5) [inline examples/website-demos/virtual-event/src/components/CentralPanel/RoomsList.tsx:6]: pull from index file

File: examples/website-demos/virtual-event/src/components/CentralPanel/RoomsList.tsx [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+
+import { useEventContext } from '../../contexts/EventContext';
+
+import './RoomsList.scss';
+
+export const RoomsList = () => {
+  const [dropdown, setDropdown] = useState(false);
```
Comment (DanC5) [inline examples/website-demos/virtual-event/src/components/CentralPanel/RoomsList.tsx:13]: make this more explicit, like `dropdownOpen`

File: examples/website-demos/virtual-event/src/components/CentralPanel/EventCard.scss [hunk 9]
```diff
@@ -76,6 +80,7 @@
     background: $light-grey;
     border-radius: 999px;
     height: fit-content;
+    color: black;
```
Comment (DanC5) [inline examples/website-demos/virtual-event/src/components/CentralPanel/EventCard.scss:83]: black is one of our statics, but good to use the variable here

File: examples/website-demos/virtual-event/src/components/CentralPanel/rooms.ts [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+import { Card7 } from '../../assets/eventCardPictures/Card7';
+
+type Room = {
+  chatType: 'main-event' | 'room';
+  content: string;
+  eventName: string;
+  label: string;
+  presenters: number;
+  title: string;
+  Image?: React.FC<{}>;
```
Comment (DanC5) [inline examples/website-demos/virtual-event/src/components/CentralPanel/rooms.ts:12]: don't need the generic

File: examples/website-demos/virtual-event/src/components/CentralPanel/rooms.ts [hunk 11]
```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import { Card1, Card2, Card4, Card5, Card6 } from '../../assets';
+import { Card7 } from '../../assets/eventCardPictures/Card7';
```
Comment (DanC5) [inline examples/website-demos/virtual-event/src/components/CentralPanel/rooms.ts:3]: why these pulled from different places?

File: examples/website-demos/virtual-event/src/components/CentralPanel/RoomsList.tsx (hunk 5 above)
Comment (yetieaterxb1) [reply examples/website-demos/virtual-event/src/components/CentralPanel/RoomsList.tsx:57]: Cant, its specific to rooms. I will implicit return though

File: examples/website-demos/virtual-event/src/components/CentralPanel/rooms.ts (hunk 3 above)
Comment (DanC5) [reply examples/website-demos/virtual-event/src/components/CentralPanel/rooms.ts:50]: that's prob a copy/paste error on francesco's side. it won't work well like this since the eventName is the same so it'll trigger the same channel. get updated info from him for one of them

File: examples/website-demos/virtual-event/src/components/CentralPanel/rooms.ts [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+    Image: Card1,
+    label: 'Private',
+    presenters: 2,
+    title: 'ESG Data - How to create it',
+    viewers: 150,
+  },
+  {
+    chatType: 'room',
+    content:
+      'How to set a business plans to use information to a competitive advantage and support enterprise goals.',
```
Comment (DanC5) [inline examples/website-demos/virtual-event/src/components/CentralPanel/rooms.ts:30]: make sure you proofread everything you write down, should be plan not plans

**Review by DanC5 — APPROVED** [review]
